opm/simulators/flow/OutputBlackoilModule.hpp

```diff
@@ -1668,6 +1678,42 @@ class OutputBlackOilModule : public GenericOutputBlackoilModule<GetPropType<Type
                 this->fip_[Inplace::Phase::CO2MassInGasPhaseMobKrg][globalDofIdx] = 0;
             }
         }
+
+        if (!this->fip_[Inplace::Phase::CO2MassInGasPhaseMaximumTrapped].empty() || 
+            !this->fip_[Inplace::Phase::CO2MassInGasPhaseMaximumUnTrapped].empty() ) {
+            Scalar sgcr = scaledDrainageInfo.Sgcr;
+            if (this->simulator_.problem().materialLawManager()->enableHysteresis()) {
+                const auto& matParams = simulator_.problem().materialLawParams(globalDofIdx);
+                // Get the maximum trapped gas saturation 
+                sgcr = MaterialLaw::trappedGasSaturation(matParams, /*maximumTrapping*/true);
+            }
+            if (!this->fip_[Inplace::Phase::CO2MassInGasPhaseMaximumTrapped].empty()) {
+                const Scalar imMobileMassGas = massGas * std::min(sgcr , sg);
+                this->fip_[Inplace::Phase::CO2MassInGasPhaseMaximumTrapped][globalDofIdx] = imMobileMassGas;
+            }
+            if (!this->fip_[Inplace::Phase::CO2MassInGasPhaseMaximumUnTrapped].empty()) {
+                const Scalar mobileMassGas = massGas * std::max(0.0, sg - sgcr);
+                this->fip_[Inplace::Phase::CO2MassInGasPhaseMaximumUnTrapped][globalDofIdx] = mobileMassGas;
+            }
+        }
+
+        if (!this->fip_[Inplace::Phase::CO2MassInGasPhaseEffectiveTrapped].empty() || 
+            !this->fip_[Inplace::Phase::CO2MassInGasPhaseEffectiveUnTrapped].empty()) {
+            Scalar sgcr = scaledDrainageInfo.Sgcr;
```

Comment:
I suggest writing this as
```C++
auto untrappedCritGas = scaledDrainageInfo.Sgcr;
```
in order to avoid "shadowing variable" warnings given that `sgcr` [is already defined above](https://github.com/OPM/opm-simulators/blob/40ec75068df36d1e1fd10b5f0e1be8d1aea87de2/opm/simulators/flow/OutputBlackoilModule.hpp#L1604).